src/Uno.UI/UI/Xaml/Media/Imaging/SvgImageSource.cs

```diff
@@ -0,0 +1,104 @@
+﻿#nullable enable
+using System;
+using System.ComponentModel;
+using System.IO;
+using System.Threading.Tasks;
+using Windows.Foundation;
+using Windows.Storage.Streams;
+using Windows.UI.Core;
+using Uno;
+
+namespace Windows.UI.Xaml.Media.Imaging
+{
+	public partial class SvgImageSource : ImageSource
+	{
+		private SvgImageSourceLoadStatus? _lastStatus;
+
+		public Uri UriSource
+		{
+			get => (Uri)GetValue(UriSourceProperty);
+			set => SetValue(UriSourceProperty, value);
+		}
+
+		public static DependencyProperty UriSourceProperty { get; } =
+			DependencyProperty.Register(
+				nameof(UriSource), typeof(Uri),
+				typeof(SvgImageSource),
+				new FrameworkPropertyMetadata(default(Uri), (s, e) => (s as SvgImageSource)?.OnUriSourceChanged(e)));
+
+		private void OnUriSourceChanged(DependencyPropertyChangedEventArgs e)
+		{
+			if (!object.Equals(e.OldValue, e.NewValue))
+			{
+				UnloadImageData();
+			}
+			InitFromUri(e.NewValue as Uri);
+#if NETSTANDARD
+			InvalidateSource();
+#endif
+		}
+
+		public SvgImageSource()
+		{
+			InitPartial();
+		}
+
+		public SvgImageSource(Uri uriSource)
+		{
+			UriSource = uriSource;
+
+			InitPartial();
+		}
+
+		internal async Task<SvgImageSourceLoadStatus> SetSourceAsync(Stream streamSource)
+		{
+			if (streamSource == null)
+			{
+				//Same behavior as windows, although the documentation does not mention it!!!
+				throw new ArgumentException(nameof(streamSource));
+			}
+
+			var copy = new MemoryStream();
+			await streamSource.CopyToAsync(copy);
+			copy.Position = 0;
+			Stream = copy;
+
+			_lastStatus = null;
+
+#if NETSTANDARD
+			await RequestOpen();
```

Comment:
That's why you made it async ... I would rather either subscribe to `ImageOpened` with a `TaskCompletionSource` (probably with a timeout), either just register a listener in `base._subscriptions` that will be invoked in the `OnOpened`. 

BTW currently you have an [async void in `ImageSource.Subscribe`](https://github.com/unoplatform/uno/pull/4589/files?diff=split&w=1#diff-53c5cc9939ff69c0778a46d886abd100ba2ef80355e9a65af72b8a2aced3a3b8R55)